map/src/manager/FavoritesManager.js

```diff
@@ -404,6 +426,164 @@ export function prepareIcon(value) {
+        url: url,
+        clienttimems: g.file.clienttimems,
+        updatetimems: g.file.updatetimems,
+        name: g.file.name,
+        hidden: g.hidden,
+    };
+    // If file data is available, process and update the favorite groups.
+    let resData = await Utils.getFileData(favGroups.mapObjs[g.name]);
+    if (resData) {
+        const favoriteFile = new File([resData], g.file.name, {
+            type: 'text/plain',
+        });
+        let favorites = await TracksManager.getTrackData(favoriteFile);
+        if (favorites) {
+            favorites.name = g.file.name;
+            // Update pointsGroups in favGroups.groups for the current group.
+            let ind = favGroups.groups.findIndex((obj) => obj.name === g.name);
+            favGroups.groups[ind].pointsGroups = favorites.pointsGroups;
+        }
+        Object.keys(favorites).forEach((t) => {
+            favGroups.mapObjs[g.name][`${t}`] = favorites[t];
+        });
+    }
+    return favGroups;
+}
+
+function createFavGroupUrl(group) {
+    return `${process.env.REACT_APP_USER_API_SITE}/mapapi/download-file?type=${encodeURIComponent(
+        group.file.type
+    )}&name=${encodeURIComponent(group.file.name)}`;
+}
+
+export async function addOpenedFavoriteGroups(files, setFavorites, setUpdateMarkers, setProcessingGroups) {
+    let newFavoritesFiles = {
+        groups: [],
+    };
+    files.forEach((file) => {
+        setProcessingGroups(true);
+        let group = FavoritesManager.createGroup(file);
+        newFavoritesFiles.groups.push(group);
+    });
+    newFavoritesFiles.mapObjs = {};
+    const newGroups = await addFavGroupsToMap(newFavoritesFiles);
+    if (newGroups) {
+        setUpdateMarkers(newGroups);
+    }
+}
+
+export function updateFavoriteGroups({
+    result,
+    selectedGroupName,
+    oldGroupName,
+    ctx,
+    useSelected = false,
+    favoriteName = null,
+    changeHidden = false,
+}) {
+    ctx.favorites.groups = FavoriteHelper.updateGroupAfterChange(
+        ctx,
+        result,
+        selectedGroupName,
+        oldGroupName,
+        changeHidden
+    );
+    let selectedGroup = ctx.favorites.groups.find((g) => g.name === selectedGroupName);
+    if (result.oldGroupResp) {
+        ctx.favorites.mapObjs[oldGroupName] = FavoriteHelper.updateGroupObj(
+            ctx.favorites.mapObjs[oldGroupName],
+            result.oldGroupResp
+        );
+    }
+    if (!ctx.favorites.mapObjs[selectedGroupName]) {
+        ctx.favorites.mapObjs[selectedGroupName] = FavoriteHelper.createGroupObj(result.newGroupResp, selectedGroup);
+    } else {
+        ctx.favorites.mapObjs[selectedGroupName] = FavoriteHelper.updateGroupObj(
+            ctx.favorites.mapObjs[selectedGroupName],
+            result.newGroupResp
+        );
+    }
+    if (useSelected && favoriteName) {
+        FavoriteHelper.updateSelectedFile(ctx, ctx.favorites, null, favoriteName, selectedGroupName, false);
+    }
+    ctx.setUpdateMarkers({ ...ctx.favorites });
+}
```

Comment:
Please use { named parameters } when number of parameters is more than 4.

updateGroupAfterChange
updateSelectedFile